waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/ThriftMetastoreClientManager.java

```diff
@@ -61,6 +71,13 @@ class ThriftMetastoreClientManager implements Closeable {
 
   private final int connectionTimeout;
   private final String msUri;
+  private final boolean impersonationEnabled;
+  private static final Duration delegationTokenCacheTtl = Duration.ofHours(1); // The default lifetime in Hive is 7 days (metastore.cluster.delegation.token.max-lifetime)
+  private static final long delegationTokenCacheMaximumSize = 1000;
+  private static final LoadingCache<DelegationTokenKey, String> delegationTokenCache = CacheBuilder.newBuilder()
```

Comment:
Is it possible to refactor this class in two one with Kerberos and one without. I'd prefer if the non kerberos path is not affected in any way by kerberos stuff like a cache that needs to be created and kept into memory.

Reply:
Yes, it should be broken down into two. It will also be simpler to maintain and ensure no issues are introduced in the non-Kerberos path.